### DISCUSSION TITLE
Using Tesla HTTP Proxy add-on for other services outside of HA

### DISCUSSION THREAD

**Discussion by alexeiw123** [llamafilm/tesla-http-proxy-addon#84] [General]
I have my add-on configured and working using a cloudflare tunnel as per the [wiki instructions](https://github.com/llamafilm/tesla-http-proxy-addon/wiki/Cloudflare-Argo-Tunnel)

Now I'm also trying to get some other services using Tesla API and I figure that they could all point to this authorised app.

The two services I use are teslamate and twcmanager.

For the time being, teslamate is still working on the legacy API, but twcmanager needs to be moved over.

The config for this is here: https://github.com/ngardiner/TWCManager/blob/2a6a914657e5414ef96987b1184333c77bc22e75/etc/twcmanager/config.json#L134-L155

```
        "teslaApiClientID": "client-id-of-registered-app",
        "teslaProxy": "https://localhost:4443",
        "teslaProxyCert": "/path/to/public_key.pem",
```

I would have thought that `teslaProxy:` would be `https://tesla.example.com`, or perhaps `homeassistant-internal.example.com` and then `teslaProxyCert:` would be `/.well-known/appspecific/com.tesla.3p.public-key.pem`, but I've not had luck trying these.

@Black2vs2 - I hope you don't mind if I tag you. I know you have a deep understanding of the cloudflare tunnel set up and this may be an easy answer for you?

Hopefully someone can help me understand this.

**Comment by llamafilm** [comment]
I think the term “public key” is misleading here, and they actually want the TLS certificate for HTTPS communication. That would be /share/tesla/selfsigned.pem.

The port is exposed externally as 4430 by default but you can change it.

**Reply by alexeiw123** [reply]
Thankyou, this is a big help! Although I still don't have it working, at least I'm trying to achieve this with the right .pem file!

Do you have any idea what the appropriate address for `teslaProxy:` would be?

In using cloudflare, my Tesla HTTP Proxy FQDN is of the format `tesla.example.com` and my NGINX domain is set to format of `homeassistant-internal.example.com`

Or would I simply use my HA server IP?

**Reply by llamafilm** [reply]
Most tools will not allow TLS on an IP address so you need to use the hostname, which is shown on the addon config page. For other addons of Home Assistant that should work out of the box. If you’re trying to reach it externally you may need to add an entry to the hosts file of your external machine.
